content/en/network_monitoring/devices/guide/build-ndm-profile.md

```diff
@@ -71,16 +72,165 @@ metrics:
+
+### Options
+
+`-f, --filters` is an option to provide the path to a YAML file containing a collection of MIB names and their list of node names to be included.
+
+For example, include `system`, `interfaces` and `ip` nodes from `RFC1213-MIB`, no node from `CISCO-SYSLOG-MIB`, and node `snmpEngine` from `SNMP-FRAMEWORK-MIB`.
+
+
+```yaml
+RFC1213-MIB:
+- system
+- interfaces
+- ip
+CISCO-SYSLOG-MIB: []
+SNMP-FRAMEWORK-MIB:
+- snmpEngine
+```
+
+**Note**: Each `MIB:node_name` correspond to exactly one and only one OID. However, some MIBs report legacy nodes that are overwritten. To resolve, edit the MIB by removing legacy values manually before loading them with this profile generator. If a MIB is fully supported, it can be omitted from the filter as MIBs not found in a filter are fully loaded. If a MIB is **not** fully supported, it can be listed with an empty node list, such as `CISCO-SYSLOG-MIB` in the example.
+
+`-a, --aliases` is an option to provide the path to a YAML file containing a list of aliases to be used as metric tags for tables, in the following format:
+
+```yaml
+aliases:
+- from:
+    MIB: ENTITY-MIB
+    name: entPhysicalIndex
+  to:
+    MIB: ENTITY-MIB
+    name: entPhysicalName
+```
+
+MIBs tables most of the time define one or more indexes, as columns within the same table, or columns from a different table and even a different MIB. The index value can be used to tag table's metrics. This is defined in the `INDEX` field in `row` nodes.
+
+As an example, `entPhysicalContainsTable` in `ENTITY-MIB` is as follows:
+
+```txt
+entPhysicalContainsEntry OBJECT-TYPE
+SYNTAX      EntPhysicalContainsEntry
+MAX-ACCESS  not-accessible
+STATUS      current
+DESCRIPTION
+        "A single container/'containee' relationship."
+INDEX       { entPhysicalIndex, entPhysicalChildIndex }  <== this is the index definition
+::= { entPhysicalContainsTable 1 }
+```
+
+or, for example, its JSON dump, where `INDEX` is replaced by `indices`:
+
+```json
+"entPhysicalContainsEntry": {
+    "name": "entPhysicalContainsEntry",
+    "oid": "1.3.6.1.2.1.47.1.3.3.1",
+    "nodetype": "row",
+    "class": "objecttype",
+    "maxaccess": "not-accessible",
+    "indices": [
+      {
+        "module": "ENTITY-MIB",
+        "object": "entPhysicalIndex",
+        "implied": 0
+      },
+      {
+        "module": "ENTITY-MIB",
+        "object": "entPhysicalChildIndex",
+        "implied": 0
+      }
+    ],
+    "status": "current",
+    "description": "A single container/'containee' relationship."
+  },
+```
+
+Indexes can be replaced by another MIB symbol that is more human-readable. You might prefer to see the interface name versus its numerical table index. This can be achieved using `metric_tag_aliases`.
+
+### Add unit tests
+
+Add a unit test in `test_profiles.py` to verify that the metric is successfully collected by the integration when the profile is enabled. These unit tests are mostly used to prevent regressions and help with maintenance.
```

Comment:
```suggestion
Add a unit test in `test_profiles.py` to verify that the integration successfully collects the metric when the profile is enabled. These unit tests are mostly used to prevent regressions and help with maintenance.
```